vision/app.py

```diff
@@ -64,50 +74,61 @@ def run(self):
 
     def do_image(self, im):
+        # Get all the contours in the image and the mask used to find them
         blobs, mask = cv_utils.get_blobs(im, self.settings['lower'], self.settings['upper'])
 
+        # Don't process any blobs if vision isn't locked
         if not self.locked:
             self.tracker.deregister_all()
             return im, mask
 
         found_blob = False
 
-        # Create array of contour areas
+        # Create list of rectangles that bound the blob
         bounding_rects = [cv2.boundingRect(blob) for blob in blobs]
 
-        # Sort array of blobs by x-value
+        # Sort list of blobs by x-value and zip them together with their corresponding bounding rectangle
         sorted_blobs = sorted(zip(bounding_rects, blobs), key=lambda x: x[0], reverse=True)
 
         goals = []
         prev_target = None
         prev_blob = None
         for bounding_rect, blob in sorted_blobs:
             if blob is not None and mask is not None:
+                # Get the location and size of the rectangle bounding the contour
                 x, y, w, h = bounding_rect
 
+                # Skip over the blob if its area is too small
                 if w * h < self.settings['min_area']:
                     continue
 
+                # Returns a rectangle of minimum area that bounds the blob (accounts for blobs at an angle)
                 target = cv2.minAreaRect(blob)
+                # Gets the coordinates of the 4 corners of the rectangle bounding the blob
                 box = np.int0(cv2.boxPoints(target))
 
+                # Straighten box to have a top-down view and get the transformed width and height
                 transformed_box = cv_utils.four_point_transform(mask, box)
                 width, height = transformed_box.shape
 
+                # Calculate the proportion of the transformed box that's filled up by the blob
                 full = cv_utils.get_percent_full(transformed_box)
                 area = width * height
 
+                # Check to make sure the box is sufficiently filled
                 if self.settings['min_area'] <= area <= self.settings['max_area'] and self.settings['min_full'] <= full <= self.settings['max_full']:
                     if self.verbose:
                         print('[Goal] x: %d, y: %d, w: %d, h: %d, area: %d, full: %f, angle: %f' % (x, y, width, height, area, full, target[2]))
 
                     if self.display:
-                        # Draw image details
+                        # Draw rectangles around goals and points on the center of the goal
                         im = cv_utils.draw_images(im, target, box)
 
                     if prev_target is not None:
+                        # Calculate the difference in angles between the previous and current target
                         sum = abs(prev_target[2]) - abs(target[2])
 
+                        # (Add the target as a goal if current target is less titled than previous target?)
```

Comment:
I was a little confused about this part of the code. Does `prev_target` refer to one piece of vision tape and the `target` refers to the one on the other side?